src/main/java/org/gridsuite/study/server/StudyService.java

```diff
@@ -242,54 +248,63 @@ private static BasicStudyInfos toBasicInfos(BasicStudyEntity entity) {
     }
 
     Mono<StudyInfos> getCurrentUserStudy(String studyName, String userId, String headerUserId) {
-        Mono<StudyEntity> studyMono = studyRepository.findStudy(userId, studyName);
-        return studyMono.flatMap(study -> {
-            if (study.isPrivate() && !userId.equals(headerUserId)) {
-                return Mono.error(new StudyException(NOT_ALLOWED));
-            } else {
-                return Mono.just(study);
-            }
-        }).map(StudyService::toInfos);
+        return getStudy(studyName, userId)
+                .switchIfEmpty(Mono.error(new StudyException(STUDY_NOT_FOUND)))
```

Comment:
why add this error ?

Reply:
it was forgotten, but it's ok it should be modified in a separate PR